### PR TITLE
daAlink work

### PR DESCRIPTION
**Pull request by TakaRikka**
still largely a wip, but a bunch of other files are touched as well so want to get this in before other work overlaps

### CODE REVIEW

File: include/JSystem/J2DGraph/J2DPane.h [hunk 1]
```diff
@@ -128,6 +128,7 @@ class J2DPane {
     void* getPointer(JSURandomInputStream* stream, u32 size, JKRArchive* archive);
     void animationTransform();
     void updateTransform(const J2DAnmTransform* transform);
+    void setUserInfo(u64 info) { mUserInfoTag = info; }
```
Comment (Julgodis) [inline include/JSystem/J2DGraph/J2DPane.h:131]: Maybe `setUserInfoTag`? If this isn't the debug version name

File: include/JSystem/J3DGraphBase/J3DMaterial.h [hunk 2]
```diff
@@ -49,6 +49,7 @@ class J3DMaterial {
     J3DColorBlock* getColorBlock() const { return mColorBlock; }
     J3DTexGenBlock* getTexGenBlock() const { return mTexGenBlock; }
     J3DDisplayListObj* getSharedDisplayListObj() const { return mSharedDLObj; }
+    J3DShape* getShape() { return mShape; }
```
Comment (Julgodis) [inline include/JSystem/J3DGraphBase/J3DMaterial.h:52]: `const` ?

File: include/d/bg/d_bg_s_acch.h [hunk 3]
```diff
@@ -137,6 +137,8 @@ class dBgS_Acch : public cBgS_Chk, public dBgS_Chk {
     void SetGroundHit() { m_flags |= GROUND_HIT; }
     void SetGroundLanding() { m_flags |= GROUND_LANDING; }
     void SetGroundAway() { m_flags |= GROUND_AWAY; }
+    u32 MaskWaterHit() { return m_flags & WATER_HIT; }
+    bool ChkWaterHit() { return MaskWaterHit(); }
```
Comment (Julgodis) [inline include/d/bg/d_bg_s_acch.h:141]: `const` ?

File: src/d/a/d_a_alink.cpp [hunk 4]
```diff
@@ -5095,50 +4911,57 @@ SECTION_RODATA u8 const daAlinkHIO_noActAtnMove_c0::m[84] = {
...
+                                                            1.3f,
+                                                            25.0f,
+                                                            10.0f,
+                                                            3.0f,
+                                                            15.0f,
+                                                            10.0f,
+                                                            12.0f,
+                                                            9.0f,
+                                                            19.0f,
+                                                            60.0f};
```
Comment (Julgodis) [inline src/d/a/d_a_alink.cpp:4931]: add `,` at the end for better formatting

File: src/d/a/d_a_alink.cpp [hunk 5]
```diff
@@ -5095,50 +4911,57 @@ SECTION_RODATA u8 const daAlinkHIO_noActAtnMove_c0::m[84] = {
...
+daAlinkHIO_sideStep_c1 const daAlinkHIO_sideStep_c0::m = {{5, 1.1f, 1.0f, 3.0f, 6.0f},
+                                                          {9, 1.0f, 0.0f, 3.0f, 2.0f},
+                                                          {6, 1.0f, 0.0f, 3.0f, 7.0f},
+                                                          {9, 1.1f, 0.0f, 3.0f, 4.0f},
+                                                          28.0f,
+                                                          22.0f,
+                                                          25.0f,
+                                                          20.0f,
+                                                          100.0f,
+                                                          10.0f};
```
Comment (Julgodis) [inline src/d/a/d_a_alink.cpp:4948]: `,`

File: src/d/a/d_a_alink.cpp [hunk 6]
```diff
@@ -5095,50 +4911,57 @@ SECTION_RODATA u8 const daAlinkHIO_noActAtnMove_c0::m[84] = {
...
+                                                    40.0f,
+                                                    3.0f,
+                                                    50.0f,
+                                                    27.0f,
+                                                    0.2f,
+                                                    12.0f,
+                                                    50.0f,
+                                                    3.0f,
+                                                    33.0f,
+                                                    1.2f};
```
Comment (Julgodis) [inline src/d/a/d_a_alink.cpp:4964]: `,`

File: src/d/a/d_a_alink.cpp [hunk 7]
```diff
@@ -5268,14 +5091,15 @@ SECTION_RODATA u8 const daAlinkHIO_cutFnJU_c0::m[84] = {
...
-COMPILER_STRIP_GATE(0x8038DB40, &daAlinkHIO_cutJump_c0::m);
+daAlinkHIO_cutJump_c1 const daAlinkHIO_cutJump_c0::m = {{17, 1.1f, 0.0f, 2.0f, 18.0f},
+                                                        {29, 1.0f, 2.0f, 0.0f, 9.0f},
+                                                        25.0f,
+                                                        27.0f,
+                                                        15.0f,
+                                                        13.0f,
+                                                        13.0f,
+                                                        1.0f,
+                                                        3.0f};
```
Comment (Julgodis) [inline src/d/a/d_a_alink.cpp:5102]: `,`

File: src/d/a/d_a_alink.cpp [hunk 8]
```diff
@@ -5398,32 +5222,49 @@ SECTION_RODATA u8 const daAlinkHIO_cutHead_c0::m[96] = {
...
+                                                                  {35, 1.25f, 0.0f, 0.0f, 22.0f},
+                                                                  20.0f,
+                                                                  33.0f,
+                                                                  2.0f,
+                                                                  6.0f,
+                                                                  13.0f,
+                                                                  0.0f,
+                                                                  10.0f,
+                                                                  250.0f,
+                                                                  5.0f};
```
Comment (Julgodis) [inline src/d/a/d_a_alink.cpp:5237]: `,`

File: src/d/a/d_a_alink.cpp [hunk 9]
```diff
@@ -5398,32 +5222,49 @@ SECTION_RODATA u8 const daAlinkHIO_cutHead_c0::m[96] = {
...
+                                                1.6f,
+                                                45.0f,
+                                                1.65f,
+                                                65.0f,
+                                                1.21f,
+                                                35.0f,
+                                                2.3f,
+                                                2.2f,
+                                                65.0f,
+                                                0.0f};
```
Comment (Julgodis) [inline src/d/a/d_a_alink.cpp:5267]: `,`

File: src/d/a/d_a_alink.cpp [hunk 10]
```diff
@@ -5773,14 +5627,21 @@ SECTION_RODATA u8 const daAlinkHIO_fmChain_c0::m[40] = {
...
+                                                              0.75f,
+                                                              0.45,
+                                                              0.6f,
+                                                              50.0f,
+                                                              10.0f,
+                                                              0.35f,
+                                                              0.3f,
+                                                              0.7f,
+                                                              0.8f,
+                                                              0.7f};
```
Comment (Julgodis) [inline src/d/a/d_a_alink.cpp:5644]: `,`

File: src/d/a/d_a_alink.cpp [hunk 11]
```diff
@@ -5858,44 +5738,53 @@ SECTION_RODATA u8 const daAlinkHIO_ironBall_c0::m[140] = {
...
-COMPILER_STRIP_GATE(0x8038EB50, &daAlinkHIO_item_c0::m);
+daAlinkHIO_item_c1 const daAlinkHIO_item_c0::m = {{12, 1.0f, 0.0f, 3.0f, 13.0f},
+                                                  {14, 1.0f, 0.0f, 3.0f, 15.0f},
+                                                  15,
+                                                  10000,
+                                                  -13000,
+                                                  6000,
+                                                  -9000,
+                                                  1.3f,
+                                                  1.3f};
```
Comment (Julgodis) [inline src/d/a/d_a_alink.cpp:5763]: `,`

**Review by Julgodis — CHANGES_REQUESTED** [review]
Good job 👍 Only had a few comments about style.

What is the `main.dol` progress at now?

I really need to figure the issue with the v-tables:
```c++
asm cXyz::~cXyz() {
// asm cXyz::~cXyz() {
extern "C" asm void __dt__4cXyzFv() {
```

File: include/JSystem/J2DGraph/J2DPane.h (hunk 1 above)
Comment (TakaRikka) [reply include/JSystem/J2DGraph/J2DPane.h:131]: yeah this is from debug

File: include/JSystem/J3DGraphBase/J3DMaterial.h (hunk 2 above)
Comment (TakaRikka) [reply include/JSystem/J3DGraphBase/J3DMaterial.h:52]: this causes some const passing errors in a few functions

File: include/JSystem/J2DGraph/J2DPane.h (hunk 1 above)
Comment (magcius) [reply include/JSystem/J2DGraph/J2DPane.h:131]: this is likely a void* and is just a random pointer for the user of J2D to throw on whatever they want

**Comment by TakaRikka** [comment]
main.dol progress is 7.66% .text, 17.37% total

**Review by Julgodis — APPROVED** [review]

**Pull request merged**